index.html

```diff
@@ -127,63 +127,42 @@
-NEL = "NEL" ":" [ directive ]  \\\*( ";" [ directive ] )
-
-directive                 = directive-name [ "=" directive-value ]
-directive-name            = token
-directive-value           = token | quoted-string
+NEL = <a>json-field</a>
+      ; See Section 2 of [[HTTP-JFV]], and Section 2 of [[RFC7159]]
         </pre>
 
-        <p>See [[!RFC7230]] section 3.2.6 for definitions of `token` and `quoted-string`. This specification defines three NEL directives: <a>report-uri</a>, <a>max-age</a>, and <a>includeSubDomains</a>. The overall requirements for directives are:</p>
-
-        <ul>
-          <li>Directive may be optional or required.</li>
-          <li>Directive names are case-insensitive.</li>
-          <li>Directive order is not significant.</li>
-          <li>Directive should be specified once.</li>
-        </ul>
+        <p>The header's value is interpreted as an array of JSON objects, as
+        described in Section 4 of [[HTTP-JFV]].</p>
 
-        <p>User agents MUST ignore any unknown or invalid directive(s), or other header field value data, that does not conform to the syntax defined in this specification. A valid <a>NEL header field</a> MUST, at a minimum, contain all of the "REQUIRED" directives defined in this specification.</p>
+        <p>Each object in the array defines an <a>NEL policy</a> for the origin.
+        The user agent MUST process the first valid policy in the array.</p>
 
-        <ul>
-          <li>If a directive is specified multiple times within the same <a>NEL header field</a>, the user agent MUST process the first declaration and ignore the rest.</li>
-          <li>If multiple valid <a>NEL header field</a>'s are present, the user agent MUST process the first valid header field and ignore the rest.</li>
-        </ul>
+        <p>User agents MUST ignore any unknown or invalid field(s) or value(s) that do not conform to the syntax defined in this specification. A valid <a>NEL header field</a> MUST, at a minimum, contain one object with all of the "REQUIRED" fields defined in this specification.</p>
 
         <p>The user agent MUST ignore the NEL header specified via a <code>meta</code> element to mitigate hijacking of error reporting via scripting attacks. The <a>NEL policy</a> MUST be delivered via the <a>NEL header field</a>.</p>
 
         <p class="note">The restriction on <code>meta</code> element is consistent with [[CSP]] specification, which restricts reporting registration to HTTP header fields only for same reasons.</p>
 
         <section>
-          <h2>The `report-uri` Directive</h2>
-          <p>The <dfn>report-uri</dfn> directive specifies a URL to which the user agent sends reports about network errors. The <a>report-uri</a> directive is a REQUIRED field to register an <a>NEL policy</a>, and OPTIONAL if the intent is to remove a previous registration - see <a>max-age</a>. The ABNF grammar for the name and value of the directive is:</p>
+          <h2>The `report-uri` Field</h2>

```

Comment:
I think we should remove report-uri entirely, to make everyone's life easier; we should assume that Reporting API exists, and use it. Practically speaking, that means the site has to specify two headers:

> Report-To: { "url": "https://example.com/reports", "group": "log-service", "max-age": 10886400}
> NEL: { "report-to": "log-service", "includeSubdomains": true, "max-age": 86400}

- We should delegate all the endpoint registration and handling to Reporting. 
- Should we allow multiple NEL policies? E.g. the "failover" use case?
- There's some interesting complications here with max-age.. 
  - What happens when report-to group's lifetime expires, while the NEL policy is still active? 
  - We could drop max-age on NEL policy and rely on Report-To group's lifecycle. However, this makes it tricky to reason about cases where you have a single report-to group associated with multiple services (e.g. NEL and CSP reports). Perhaps two separate lifetimes is OK; we just need to think through the possible workflows.